llama_index/llms/openai_utils.py

```diff
@@ -288,15 +288,18 @@ def to_openai_function(pydantic_class: Type[BaseModel]) -> Dict[str, Any]:
 
     Convert pydantic class to OpenAI function.
     """
-    return to_openai_tool(pydantic_class)
+    return to_openai_tool(pydantic_class, description=None)
 
 
-def to_openai_tool(pydantic_class: Type[BaseModel]) -> Dict[str, Any]:
+def to_openai_tool(
+    pydantic_class: Type[BaseModel], description: Optional[str] = None
+) -> Dict[str, Any]:
     """Convert pydantic class to OpenAI tool."""
     schema = pydantic_class.schema()
+    schema_description = schema.get("description", None) or description
```

Comment:
should we put `description` before `schema.get("description", None)` when assign schema_description

Reply:
Ultimately it shouldn't matter because only one of both should be provided (in the `OpenAIPydanticProgram` class, there is a `_description_eval` method that ensures that either one or the other (but not both) is defined).
Hope that answers your question.